app/topics/topic-management/topic-input/topic-input.component.ts

```diff
@@ -0,0 +1,162 @@
+import { User } from '../../../core/user/user.model';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+
+
+@Component({
+  selector: 'hip-topic-input',
+  templateUrl: './app/topics/topic-management/topic-input/topic-input.component.html',
+  styleUrls: ['./app/topics/topic-management/topic-input/topic-input.component.css'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => TopicInputComponent),
+    multi: true
+  }]
+})
+export class TopicInputComponent implements OnInit {
+  @Input() depthLeft = 0;
+  @Input() depth = 0;
+  @Input() showContent = true;
+  @Input() topic: Topic = Topic.emptyTopic();
+  @Input() currentParent: Topic = null;
+  @Output() modelChange = new EventEmitter<Topic>();
+
+  updateData() {
+    this.modelChange.emit(this.topic);
+  }
+
+  students = '';
+  subTopics: Topic[] = this.topic.subTopics;
+  playAnimation = !this.showContent;
+
+  constructor(private topicService: TopicService,
+              private router: Router,
+              private route: ActivatedRoute,
+              private toasterService: ToasterService) {
+  }
+
+  ngOnInit() {
+    if (this.topic.deadline !== null) {
+      this.topic.deadline = this.topic.deadline.slice(0, 10);
+    }
+    this.playAnimation = !this.showContent;
+    if (this.route.snapshot.url[0].path === 'topics') {
+      let id = +this.route.snapshot.params['id']; // (+) converts string 'id' to a number
+      this.topicService.getTopic(id).then(
+        response => {
+          this.topic = <Topic> response;
+          if (this.topic.deadline !== null) {
+            this.topic.deadline = this.topic.deadline.slice(0, 10);
+          }
+        }
+      ).catch(
+        error => this.toasterService.pop('error', 'Error fetching topic', error)
+      );
+      this.topicService.getStudentsOfTopic(id).then(
+        response => this.topic.students = <User[]> response
+      ).catch(
+        error => this.toasterService.pop('error', 'Error fetching Students', error)
+      );
+      this.topicService.getReviewersOfTopic(id).then(
+        response => this.topic.reviewers = <User[]> response
+      ).catch(
+        error => this.toasterService.pop('error', 'Error fetching Reviewers', error)
+      );
+      this.topicService.getSupervisorsOfTopic(id).then(
+        response => this.topic.reviewers = <User[]> response
+      ).catch(
+        error => this.toasterService.pop('error', 'Error fetching Supervisors', error)
+      );
+      this.topicService.getSubTopics(id).then(
+        response => this.topic.subTopics = <Topic[]> response
+      ).catch(
+        error => this.toasterService.pop('error', 'Error fetching SubTopics', error)
+      );
+    }
+    console.log(this.topic);
+  }
+
+  toggleContent() {
+    this.showContent = !this.showContent;
+  };
+
+  addSubTopic() {
+    this.subTopics.push(Topic.emptyTopic([this.topic]));
+  }
+
+  addTopic() {
+
+  }
+
+  removeTopic() {
```

Comment:
thats me :P 
So, you can remove all the functions, they are leftovers from reordering the topic module, and splitting up edit / create / show topic